common-npm-packages/docker-common-v2/registryauthenticationprovider/registryauthenticationtoken.ts

```diff
@@ -55,13 +55,23 @@ export default class RegistryServerAuthenticationToken {
     }
 }
 
-export function getDockerRegistryEndpointAuthenticationToken(endpointId: string): RegistryServerAuthenticationToken {
+export async function getDockerRegistryEndpointAuthenticationToken(endpointId: string): Promise<RegistryServerAuthenticationToken> {
```

Comment:
Can you please cover this with some unit tests ?

Reply:
Not quite sure I understand. The change here only affects managed identity which can only be tested on a VM. The current SPN has unit tests which are inside the L0 file of the DockerV2 task. These tests all call the docker.ts file which is where the getDockerRegistryEndpointAuthenticationToken method is used. If you would like me to add more unit tests, I can do so, but this function is already being tested there with each call of a Docker Registry.